rviz_common/src/rviz_default_plugins/point_cloud_common.cpp

```diff
@@ -0,0 +1,1038 @@
+      "Rendering mode to use, in order of computational complexity.",
+      display_, SLOT(updateStyle()), this);
+  style_property_->addOption("Points", rviz_rendering::PointCloud::RM_POINTS);
+  // TODO(greimela): Enable as soon as shaders work again
+//  style_property_->addOption( "Squares", rviz_rendering::PointCloud::RM_SQUARES );
+//  style_property_->addOption( "Flat Squares", rviz_rendering::PointCloud::RM_FLAT_SQUARES );
+//  style_property_->addOption( "Spheres", rviz_rendering::PointCloud::RM_SPHERES );
+//  style_property_->addOption( "Boxes", rviz_rendering::PointCloud::RM_BOXES );
+
+  point_world_size_property_ = new rviz_common::properties::FloatProperty("Size (m)", 0.01,
+      "Point size in meters.",
+      display_, SLOT(updateBillboardSize()), this);
+  point_world_size_property_->setMin(0.0001);
+
+  point_pixel_size_property_ = new rviz_common::properties::FloatProperty("Size (Pixels)", 3,
+      "Point size in pixels.",
+      display_, SLOT(updateBillboardSize()), this);
+  point_pixel_size_property_->setMin(1);
+
+  alpha_property_ = new rviz_common::properties::FloatProperty("Alpha", 1.0,
+      "Amount of transparency to apply to the points.  Note that this is experimental "
+      "and does not always look correct.",
+      display_, SLOT(updateAlpha()), this);
+  alpha_property_->setMin(0);
+  alpha_property_->setMax(1);
+
+  decay_time_property_ = new rviz_common::properties::FloatProperty("Decay Time", 0,
+      "Duration, in seconds, to keep the incoming points.  0 means only show the latest points.",
+      display_, SLOT(queueRender()));
+  decay_time_property_->setMin(0);
+
+  xyz_transformer_property_ = new rviz_common::properties::EnumProperty("Position Transformer", "",
+      "Set the transformer to use to set the position of the points.",
+      display_, SLOT(updateXyzTransformer()), this);
+  connect(xyz_transformer_property_, SIGNAL(requestOptions(
+      rviz_common::properties::EnumProperty *)),
+    this, SLOT(setXyzTransformerOptions(rviz_common::properties::EnumProperty *)));
+
+  color_transformer_property_ = new rviz_common::properties::EnumProperty("Color Transformer", "",
+      "Set the transformer to use to set the color of the points.",
+      display_, SLOT(updateColorTransformer()), this);
+  connect(color_transformer_property_,
+    SIGNAL(requestOptions(rviz_common::properties::EnumProperty *)),
+    this, SLOT(setColorTransformerOptions(rviz_common::properties::EnumProperty *)));
+}
+
+void PointCloudCommon::initialize(
+  rviz_common::DisplayContext * context,
+  Ogre::SceneNode * scene_node)
+{
+//  transformer_class_loader_ = new pluginlib::ClassLoader<PointCloudTransformer>( "rviz",
+// "rviz::PointCloudTransformer" );
+  loadTransformers();
+
+  context_ = context;
+  scene_node_ = scene_node;
+
+  updateStyle();
+  updateBillboardSize();
+  updateAlpha();
+  updateSelectable();
+
+//  spinner_.start();
```

Comment:
Please add:

```
// TODO(wjwwood): replace with executor when callback groups can be added to executors
```

Reply:
Would this be necessary at all? I would actually remove it for now, since everything works fine with the executors we added in the visualization manager.

Reply:
Well, this code was originally expecting to have its own thread, so to emulate that we would need the changes to the executor I was planning. Maybe we don't end up needing it, but I think it might be important.